pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/security/ApexDangerousMethodsRule.java

```diff
@@ -35,14 +50,49 @@ public Object visit(ASTUserClass node, Object data) {
             return data;
         }
 
+        collectBenignVariables(node);
```

Comment:
be careful here. A single rule instance will be used to analyze multiple files. That means the `whiteListedVariables` will accumulate all names from previous files and could produce false negatives. You should either clear the set for each class, or move it to a local variable.